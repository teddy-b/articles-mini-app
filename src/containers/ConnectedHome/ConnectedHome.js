import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import {
  loadArticles,
  selectArticle,
  loadComments,
  loadReplies,
  addComment,
  addReply,
  nextPage,
  prevPage
} from '../../actions';
import Home from '../../components/App/Home';

const mapStateToProps = state => ({
  articles: state.articles.data,
  page: state.page,
  articlesCount: state.articles.totalCount,
  comments: state.comments,
  fetching: state.fetching,
  selectedArticleId: state.selectedArticleId
});

const mapDispatchToProps = dispatch => ({
  onLoadArticles() {
    dispatch(loadArticles());
  },
  onLoadComments(articleId) {
    dispatch(loadComments(articleId));
  },
  onLoadReplies(parentCommentId) {
    dispatch(loadReplies(parentCommentId));
  },
  onSelectArticle(articleId) {
    dispatch(selectArticle(articleId));
  },
  onAddComment(articleId, event) {
    if (event.key === 'Enter') {
      dispatch(addComment(articleId, event.target.value));
      event.target.value = '';
      event.target.blur();
    }
  },
  onAddReply(parentCommentId, text) {
    dispatch(addReply(parentCommentId, text));
  },
  onNextPage() {
    dispatch(nextPage());
  },
  onPrevPage() {
    dispatch(prevPage());
  }
});

const ConnectedHome = withRouter(connect(mapStateToProps, mapDispatchToProps)(Home));

export default ConnectedHome;
