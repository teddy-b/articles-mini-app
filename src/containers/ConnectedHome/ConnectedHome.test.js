import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import renderer from 'react-test-renderer';

import configureMockStore from 'redux-mock-store';
import ConnectedHome from './ConnectedHome';

describe('App', () => {
  const storeMock = configureMockStore([thunk]);
  const store = storeMock({
    articles: {
      data: [],
      totalCount: 0
    },
    page: 0,
    comments: [],
    fetching: false,
    selectedArticleId: null,
    errors: []
  });
  const props = {
    articles: [],
    articlesCount: 0,
    comments: [],
    fetching: false,
    match: {
      params: {
      }
    },
    onAddComment: jest.fn(),
    onAddReply: jest.fn(),
    onLoadArticles: jest.fn(),
    onLoadComments: jest.fn(),
    onLoadReplies: jest.fn(),
    onNextPage: jest.fn(),
    onPrevPage: jest.fn(),
    onSelectArticle: jest.fn(),
    page: 0,
    selectedArticleId: null
  };

  it('renders correctly', () => {
    const component = renderer.create(
      <MemoryRouter>
        <ConnectedHome
          {...props}
          store={store}
        />
      </MemoryRouter>
    );

    expect(component).toMatchSnapshot();
  });
});
