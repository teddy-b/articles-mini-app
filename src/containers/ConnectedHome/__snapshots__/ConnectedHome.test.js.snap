// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="home"
>
  <div
    className="articles-list"
  >
    <h2>
      Articles list
    </h2>
    <div
      className="pagination"
    />
  </div>
  <div
    className="articles-details-empty"
  >
    <h2>
      Nothing to show
    </h2>
    <div>
      Select article from the list to see details
    </div>
  </div>
</div>
`;
